app/src/main/java/com/example/myapplication/model/Product.kt

```diff
@@ -11,4 +14,45 @@ data class Product(
     var reviewCount: Int = 0,
     var ratingValue: Double = 0.0,
     var points: Int = 0
-)
\ No newline at end of file
+): Parcelable {
+    constructor(parcel: Parcel) : this(
```

Comment:
You can get rid off all this generated methods that Parcelable generates you here. Use kotlinx.parcelize.Parcelize annotation on Product class, something like this ->

@Parcelize
data class Product(...): Parcelable